File: src/MixNetFx/Facesso.Interfaces/Jensen/JensenProductionDataImportClasses.vb
Imports System.Data.OleDb

<FacessoImportFilterName("Jensen Produktionsdatenimport", FacessoImportType.WorkGroupData, FacessoInterfaceBrand.KannegiesserProductionData)> _
Public Class JensenProductionDataImportTaskElement
    Inherits FacessoProductionDataImportTaskItemBase

    Private myJensenSQLConnectionString As String
    Private myJensenDeviceID As String
    Private myCurrDate As Date
    Private myCurrOrgData As DataTable
    Private myCurrFacData As ProductionDataTable

    Public Property JensenSQLConnectionString() As String
        Get
            Return myJensenSQLConnectionString
        End Get
        Set(ByVal value As String)
            myJensenSQLConnectionString = value
        End Set
    End Property

    Public Property JensenDeviceID() As String
        Get
            Return myJensenDeviceID
        End Get
        Set(ByVal value As String)
            myJensenDeviceID = value
        End Set
    End Property

    Public Overrides Function ConfigureImportFilter() As System.Windows.Forms.DialogResult
        Dim locFrm As New frmJensenProdDataConfigDialog
        Return locFrm.HandleDialog(Me)
    End Function

    Public Overrides Function GetData(ByVal ProductionDate As Date, ByVal Shift As ShiftCombination) As IImportResultTable
        retrieveDataForDate(ProductionDate)
        Return myCurrFacData
    End Function

    Public Overrides ReadOnly Property ImportType() As FacessoImportType
        Get
            Return FacessoImportType.WorkGroupData
        End Get
    End Property

    Public Overrides ReadOnly Property InterfaceBrand() As FacessoInterfaceBrand
        Get
            Return FacessoInterfaceBrand.JensenProductionData
        End Get
    End Property

    Public Overrides Function ConfigureGenericInterface() As System.Windows.Forms.DialogResult
        Return Windows.Forms.DialogResult.OK
    End Function

    Public Overrides ReadOnly Property IsGenericInterfaceConfigured() As Boolean
        Get
            Return True
        End Get
    End Property

    Public Overrides Function AssembleConversionItems() As FacessoConversionItemsBase
        Return MyBase.AssembleConversionItems
    End Function
End Class

File: src/NetFxCSharp/Facesso.Interfaces/Jensen/JensenProductionDataImportClasses.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Data.OleDb;
using System.Diagnostics;
using System.Linq;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace Facesso.Interfaces
{
    [FacessoImportFilterName("Jensen Produktionsdatenimport", FacessoImportType.WorkGroupData, FacessoInterfaceBrand.KannegiesserProductionData)]
    public partial class JensenProductionDataImportTaskElement : FacessoProductionDataImportTaskItemBase
    {
        private string myJensenSQLConnectionString;
        private string myJensenDeviceID;
        private System.DateTime myCurrDate;
        private DataTable myCurrOrgData;
        private ProductionDataTable myCurrFacData;
        public string JensenSQLConnectionString
        {
            get
            {
                return myJensenSQLConnectionString;
            }

            set
            {
                myJensenSQLConnectionString = value;
            }
        }

        public string JensenDeviceID
        {
            get
            {
                return myJensenDeviceID;
            }

            set
            {
                myJensenDeviceID = value;
            }
        }

        public override System.Windows.Forms.DialogResult ConfigureImportFilter()
        {
            frmJensenProdDataConfigDialog locFrm = new frmJensenProdDataConfigDialog();
            return locFrm.HandleDialog(this);
        }

        public override IImportResultTable GetData(System.DateTime ProductionDate, ShiftCombination Shift)
        {
            retrieveDataForDate(ProductionDate);
            return myCurrFacData;
        }

        public override FacessoImportType ImportType
        {
            get
            {
                return FacessoImportType.WorkGroupData;
            }
        }

        public override FacessoInterfaceBrand InterfaceBrand
        {
            get
            {
                return FacessoInterfaceBrand.JensenProductionData;
            }
        }

        public override System.Windows.Forms.DialogResult ConfigureGenericInterface()
        {
            return System.Windows.Forms.DialogResult.OK;
        }

        public override bool IsGenericInterfaceConfigured
        {
            get
            {
                return true;
            }
        }

        public override FacessoConversionItemsBase AssembleConversionItems()
        {
            return base.AssembleConversionItems();
        }
    }
}